### PR TITLE
feat(dashboard): populate all repositories

### PR DESCRIPTION
**Pull request by olarclara**
## What kind of change does this PR introduce?

Fetch teams' v2 repositories (projects) and display them in list and grid view.

## What is the new behavior?

|List|Grid|
|:-:|:-:|
|m9mgp6-3000 preview csb app_dashboard_synced-sandboxes_workspace=b5123bcc-da66-446f-88e2-5b89c9c37d27 (1)|m9mgp6-3000 preview csb app_dashboard_synced-sandboxes_workspace=b5123bcc-da66-446f-88e2-5b89c9c37d27|

## What steps did you take to test this? This is required before we can merge, make sure to test the flow you've updated.

1. Open the dashboard
2. Navigate to `All repositories`
3. See the team's repositories listed on the screen

## Missing things

- Loading state for the grid view: replace by a single square.
- Incremental fetching to display data faster.

### CODE REVIEW

**Comment by lbogdan** [comment]
Build for latest commit https://github.com/codesandbox/codesandbox-client/pull/6917/commits/705b87188edf261348dedf68281cc8bb4cf6cfa6 is at https://pr6917.build.csb.dev/s/new.

File: packages/app/src/app/overmind/namespaces/dashboard/state.ts [hunk 1]
```diff
@@ -65,8 +66,8 @@ export type State = {
     older: Sandbox[];
   };
   curatedAlbums: DashboardAlbum[];
-  // TODO: fix unknown
   contributions: DashboardContributionBranch[] | null;
+  v2Repositories: DashboardV2Repository[] | null;
```
Comment (alexnm) [inline packages/app/src/app/overmind/namespaces/dashboard/state.ts:70]: should we call them `repositories` to not carry the v2 thing everywhere?

File: packages/app/src/app/pages/Dashboard/types.ts [hunk 2]
```diff
@@ -140,6 +141,13 @@ export type DashboardContributionBranch = {
   >['recentBranches'][number];
 };
 
+export type DashboardV2Repository = {
+  type: 'v2-repo';
+  repo: NonNullable<
```
Comment (alexnm) [inline packages/app/src/app/pages/Dashboard/types.ts:146]: I would export a `Repository` type from somewhere so we can reuse based on the graphql generated typing

File: packages/app/src/app/pages/Dashboard/Components/V2Repository/index.tsx [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+          ':focus-visible': {
+            boxShadow: '0 0 2px 1px rgba(255, 255, 255, 0.4)',
+          },
+        })}
+        direction="vertical"
+        gap={4}
+        onClick={handleClick}
+        onKeyDown={e => e.key === 'Enter' && handleClick(e)}
+        // TODO: refine semantics when/if the context menu gets implemented.
+        role="link"
```
Comment (alexnm) [inline packages/app/src/app/pages/Dashboard/Components/V2Repository/index.tsx:58]: let's do a quick chat with the design team about the interaction model in the dashboard. `branches` and `repos` are links, while the sandboxes are selectable via one click and are open by dbl click. wonder if we should mimic that or disable selection for the pages where we show branch and repo cards

**Review by alexnm — APPROVED** [review]
Left a few comments, but it looks good

File: packages/app/src/app/overmind/namespaces/dashboard/state.ts (hunk 1 above)
Comment (olarclara) [reply packages/app/src/app/overmind/namespaces/dashboard/state.ts]: Removed `v2` references in commit [14c8fa7](https://github.com/codesandbox/codesandbox-client/pull/6917/commits/14c8fa7fe40935878d890de0e5008a6003b18ec7).

File: packages/app/src/app/pages/Dashboard/types.ts (hunk 2 above)
Comment (olarclara) [reply packages/app/src/app/pages/Dashboard/types.ts]: Created a fragment and used it as the type in commit: [14c8fa7](https://github.com/codesandbox/codesandbox-client/pull/6917/commits/14c8fa7fe40935878d890de0e5008a6003b18ec7).

File: packages/app/src/app/pages/Dashboard/Components/V2Repository/index.tsx (hunk 3 above)
Comment (olarclara) [reply packages/app/src/app/pages/Dashboard/Components/V2Repository/index.tsx]: Will be addressed in https://linear.app/codesandbox/issue/XTD-140/recent-page-improve-interaction-model.

**Pull request merged**